[Li Xintong]iP

Emrysil opened the pull request:
# SebastianMax
> Hello World!

SebastianMax frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is:
1. download it from [here](https://github.com/Emrysil/ip/releases/tag/A-Jar)
2. double click it
3. add your tasks
4. let it manage your tasks for you 💯 
   
And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a programmer, you can use it to practice Java too! Here is the `Main` method
```Java
public static void main(String[] args) {
    System.out.println("Hello World");
}
```

Commit:
Add Gradle support

Emrysil committed:
implementation of Greet, Echo and Exit

Emrysil committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

Emrysil committed:
Add the ability to mark tasks as done and undone.

Emrysil committed:
Add support for tracking three types of tasks: Todos, Deadlines and Events

Emrysil committed:
Introduced some custom exceptions specific to Sebastian and added exception handling mechanisms

Emrysil committed:
Added support for deleting tasks from the list

Emrysil committed:
Added a few Exception classes, fixed small bugs

Emrysil committed:
Fixed some bugs and added documentation

Emrysil committed:
Added auto testing of text ui

Emrysil committed:
Added the functionality to save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when Duke starts up.

Emrysil committed:
Added the functionality such that Sebastian now understand dates and times. Sebastian is now able to return deadlines/event occurring on a specific date.

Emrysil committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Added the functionality such that Sebastian now understand dates and times. Sebastian is now able to return deadlines/event occurring on a specific date.

# Conflicts:
#	src/main/java/Duration.java
#	src/main/java/EndTime.java
#	src/main/java/Sebastian.java
#	src/main/java/TaskList.java
#	src/main/java/task/Deadline.java
#	src/main/java/task/Event.java
#	src/main/java/task/Todo.java

Emrysil committed:
removed irrelevant file

Emrysil committed:
Refactored the code to extract out closely related code as classes and grouped related classes into packages

Emrysil committed:
Merge branch 'add-gradle-support' of https://github.com/Emrysil/ip
Use gradle to automate some of the build tasks of the project

Emrysil committed:
Added gradle and some JUnit tests

Emrysil committed:
A-Jar

Emrysil committed:
Revert "A-Jar"

This reverts commit fa059329f690db381a80a8db25648ebf1367c31c.

Emrysil committed:
Revert "Revert "A-Jar""

This reverts commit 125c84ad13198e8d39101b212b351f684bf019a8.

Emrysil committed:
Fixed small format problem

Emrysil committed:
Merge commit '63c187e5de4bff72e4a9704b1125ae797a91fa3b'

Emrysil committed:
added javadoc

Emrysil committed:
Fixed some styling to comply with coding standard

Emrysil committed:
Added support for users to find a task by searching for a keyword

Emrysil committed:
Merge branch 'branch-A-CodingStandard'

Emrysil committed:
Merge branch 'branch-A-JavaDoc'

Emrysil committed:
Modified code to follow the style guide. Added GUI

Emrysil committed:
Removed test-ui test

Emrysil committed:
fixed small bug in GUI

zxisatree commented on src/main/java/controllers/DialogBox.java:
```diff
@@ -0,0 +1,109 @@
+    /**
+     * Styles the dialog box and other components it contains
+     */
+    private void style() {
```
Perhaps this method could be named applyStyle instead? style could be a noun as well, and the meaning of this method might be ambiguous.

zxisatree commented on src/main/java/controllers/Main.java:
```diff
@@ -0,0 +1,34 @@
+import javafx.stage.Stage;
+import sebastian.Sebastian;
+
+
```
Is more than one line of whitespace necessary here?

zxisatree commented on src/main/java/sebastian/command/ExitCommand.java:
```diff
@@ -0,0 +1,28 @@
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+        String res = "Bye. It's my pleasure to serve you";
```
The variable `res` seems unneccessary, as it is only used once. Might be simpler to just return the string directly.

zxisatree commented on src/main/java/sebastian/main/Parser.java:
```diff
@@ -0,0 +1,55 @@
+public class Parser {
+
+    /**
+     * To interpret user command and dispatch the job to different command handlers for execution
```
Perhaps "Interprets user commands..." instead of "To interpret user command..." is more descriptive and follows the imperative tone that the coding standard suggests.

zxisatree commented on src/main/java/sebastian/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * Checked if the deadline is on the same day as the provided date
```
"Checks if..." instead of "Checked if..." could be used to follow the coding standard's tone.

zxisatree commented on src/main/java/sebastian/time/DatePattern.java:
```diff
@@ -0,0 +1,21 @@
+
+    private final String format;
+
+    DatePattern(String format) {
```
Missing access modifier? Not sure if the constructor is intended to be default.

zxisatree reviewed (commented):
I like the separation of code into the various packages, makes it very easy to navigate and find relevant code. Overall very neat and comprehensive code.

Emrysil committed:
minor modifications to the GUI

Emrysil committed:
resolved conflict

Emrysil committed:
fixed some code style problems

Emrysil committed:
Add assertions

Add assertions to various command classes to ensure the success of
add and delete operation before writing to the hard disk.

Emrysil committed:
Improve code quality

Reduce error-prone practices by using the final else in if-else
construct for error detection only.

Improve code readability by avoiding magic numbers.

Emrysil committed:
Use Streams

Use Streams to find tasks based on user-supplied keywords.

Emrysil committed:
Merge pull request #1 from Emrysil/branch-A-Assertions

Add assertions

Emrysil committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/sebastian/command/AddDeadlineCommand.java
#	src/main/java/sebastian/command/AddEventCommand.java
#	src/main/java/sebastian/command/AddTodoCommand.java

Emrysil committed:
Merge branch 'master' into branch-A-Streams

Emrysil committed:
Merge pull request #2 from Emrysil/branch-A-CodeQuality

Improve code quality

Emrysil committed:
Resolve conflict

Resolve conflict when merging master into A-Streams

Emrysil committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/sebastian/command/GetCommand.java

Emrysil committed:
Merge pull request #3 from Emrysil/branch-A-Streams

Use Streams

Jxleejiaxin commented on src/main/java/sebastian/main/Storage.java:
```diff
@@ -0,0 +1,87 @@
+     */
+    public TaskList formTaskListFromData() throws CannotLoadDataException {
+        TaskList taskList = new TaskList();
+        File file = new File("src/main/resources/SebastianData.txt");
```
Perhaps it might be better to extract out the file path for your Storage class methods? Your Storage class constructor receives a String _filePath_ as argument and creates a _file_ field but it is immediately modified in your methods. It is also more convenient if you decide to change your file path in the future.

Emrysil committed:
Support updating task details

Add functionality to support updating task details including task
description, start time and end time

Jxleejiaxin reviewed (commented):
I like how you have extracted so many different kinds of exceptions. I think more system outputs could be extracted to the UI class. Keep up the good work!

Emrysil committed:
Add github actions

Emrysil committed:
Add user guide

Add one entry to the user guide to include syntax for updating details
of a task.

Emrysil committed:
Fix style violations

Fix style violations spotted by checkstyle.

Emrysil committed:
Fix bugs

Fix some bugs inside the Task class that is causing task completion
status to be read wrongly from the disk.

Emrysil committed:
Add Personality

Created a unique personality for Sebastian

Emrysil committed:
Add a brief User Guide

Added a brief User Guide for Sebastian

Emrysil committed:
no message

Emrysil committed:
Fix minor bugs

Emrysil committed:
Modify storage

Instead of saving data to SebastianData.txt, save to
data/SebastianData.txt

Emrysil committed:
Fix style violations
